### DISCUSSION TITLE
How to silence the ExperimentalWarning?

### DISCUSSION THREAD

**Discussion by psarka** [dagster-io/dagster#13112] [Q&A] [answered]
I'm unable to cleanly silence the ExperimentalWarning :( I'm using MultiPartitionsDefinition, and it dumps the warnings non-stop on my screen (I think on every definitions update, but I'm not sure).

I added the following two lines on top of every file of my repository, still warnings appear:
```
import warnings
warnings.simplefilter("ignore")
```

Based on the warning text itself, it originates somewhere from deep down dagster core:

```
[...]\lib\site-packages\dagster\_core\host_representation\external_data.py:681: 
ExperimentalWarning: "MultiPartitionsDefinition" is an experimental class. It may break in future versions, even between dot releases.
To mute warnings for experimental functionality, invoke warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning)
or use one of the other methods described at https://docs.python.org/3/library/warnings.html#describing-warning-filters.
  return MultiPartitionsDefinition(
```

The warnings do go away if I start the interpreter with `PYTHONWARNINGS='ignore'` flag, but I'd rather not do that.

**Comment by jamiedemaria** [comment]
hey @psarka  I've tried to replicate this, but when I add 

```
import warnings
warnings.simplefilter("ignore")
```
to the top of my files, experimental warnings are silenced. Can you provide some more information about how you are running dagster? Are you just starting the UI locally (using `dagster dev` or `dagit`), or some other deployment method?

**Comment by sryza** [comment]
btw I don't think this needs to be marked experimental anymore: https://github.com/dagster-io/dagster/pull/13123

**Comment by psarka** [comment]
Very happy to see this feature exit Experimental status! Anyway, @jamiedemaria, if it still makes sense to look into this for the future warnings, here is a repro that triggers the warning, even though warnings are ignored in every file.
https://github.com/psarka/repro

I'm using 1.1.21 (latest available on conda forge), Windows 10.

**Comment by f-tremblay** [comment]
You can try this:
``` 
import warnings
import dagster.ExperimentalWarning
warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning)
```
It worked for me.

**Comment by joscha** [comment, marked as answer]
in later versions (for example in `1.10.4`) this becomes:
```py
warnings.filterwarnings("ignore", category=dagster.BetaWarning)
```

**Reply by garethbrickman** [reply]
This is now formally documented here: [Filtering API lifecycle warnings](https://docs.dagster.io/api/api-lifecycle/filtering-api-lifecycle-warnings)
